selfservice/src/main/java/no/helseid/selfservice/clientsecret/DefaultClientSecretUpdater.java

```diff
@@ -0,0 +1,82 @@
+package no.helseid.selfservice.clientsecret;
+
+import com.nimbusds.jose.jwk.JWK;
+import no.helseid.dpop.DPoPProofCreator;
+import no.helseid.endpoints.token.AccessTokenResponse;
+import no.helseid.endpoints.token.TokenRequestDetails;
+import no.helseid.endpoints.token.TokenResponse;
+import no.helseid.exceptions.HelseIdException;
+import no.helseid.grants.ClientCredentials;
+import no.helseid.selfservice.endpoints.clientsecret.ClientSecretEndpoint;
+import no.helseid.selfservice.endpoints.clientsecret.ClientSecretResponse;
+import no.helseid.selfservice.endpoints.clientsecret.ClientSecretSuccessResponse;
+import no.helseid.signing.Algorithm;
+import no.helseid.signing.RSAKeyReference;
+import no.helseid.signing.Util;
+
+import java.net.URI;
+import java.util.List;
+
+
+/**
+ * Default implementation of a ClientSecretUpdater
+ */
+public class DefaultClientSecretUpdater implements ClientSecretUpdater {
+  private final URI selfServiceClientSecretEndpoint;
+  private final ClientCredentials clientCredentials;
+  private final DPoPProofCreator dPoPProofCreator;
+  private final List<String> clientSecretScope;
+
+  /**
+   * Creates the default implementation of the ClientSecretUpdater
+   * @param selfServiceClientSecretEndpoint the full endpoint for updating client secrets
+   * @param clientCredentials a client credentials instance for the client
+   * @param clientSecretScope the scope(s) required for the selfServiceClientSecretEndpoint
+   */
+  public DefaultClientSecretUpdater(
+      final URI selfServiceClientSecretEndpoint,
+      final ClientCredentials clientCredentials,
+      final List<String> clientSecretScope
+  ) {
+    this.selfServiceClientSecretEndpoint = selfServiceClientSecretEndpoint;
+    this.clientCredentials = clientCredentials;
+    this.dPoPProofCreator = clientCredentials.getCurrentDPoPProofCreator();
+    this.clientSecretScope = clientSecretScope;
+  }
+
+  public UpdatedClientSecretSuccess generateNewClientSecret() throws HelseIdException {
+    UpdatedClientSecretResult result = updateClientSecret();
+
+    if (result instanceof UpdatedClientSecretSuccess updatedClientSecretSuccess) {
+      return updatedClientSecretSuccess;
+    }
+
+    throw new HelseIdException("Update of client secret failed");
+  }
+
+  public UpdatedClientSecretResult updateClientSecret() throws HelseIdException {
+    TokenRequestDetails requestDetails = new TokenRequestDetails.Builder()
+        .addMultipleScope(clientSecretScope)
+        .build();
+    TokenResponse tokenResponse = clientCredentials.getAccessToken(requestDetails);
+
+    if (tokenResponse instanceof AccessTokenResponse accessTokenResponse) {
```

Comment:
@kennethmyhra 

Tok tipset ditt of snudde om på logikken her:
Sjekker om det er en suksess, hvis det ikke er suksess tolker vi det som en "error"
Både for TokenResponse på linje 63 og ClientSecretResponse på linje 73